paginated_reader.go

```diff
@@ -0,0 +1,81 @@
+package timber
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/peterhellberg/link"
+	"github.com/pkg/errors"
+)
+
+// paginatedReadCloser implements the io.ReadCloser interface, wrapping the
+// io.ReadCloser from an HTTP response. The reader will first exhaust the
+// underlying reader, then, using the HTTP response headers, request the next
+// page, if any, and replace the underlying reader. An io.EOF error is returned
+// when either the underlying reader is exhausted and the HTTP response header
+// does not contain a "next" field, or the "next" field's URL returns no data.
```

Comment:
Can these comments about how `Read` behaves be moved to the `Read` method, where they're the most relevant?